### ISSUE TITLE
Handle status response with a Class during sevice readiness check

### ISSUE THREAD

**Issue by tianleh** [opensearch-project/opensearch-build#1217]
### Is your feature request related to a problem? Please describe

#604 

A follow up of @dblock 's PR comment https://github.com/opensearch-project/opensearch-build/pull/1166#discussion_r757802692

### Describe the solution you'd like

handle the json with Class

**Comment by ohltyler** [comment]
@tianleh even better, what about using the `number_of_pending_tasks` field to determine cluster readiness? For example, I've seen integ tests occasionally fail due to missing indices, and looks like the reported cluster health has a `number_of_pending_tasks` > 0 (see [here](https://github.com/opensearch-project/opensearch-build/issues/1531#issuecomment-1018779250))

**Comment by tianleh** [comment]
@ohltyler Totally make sense! I am focusing on https://github.com/opensearch-project/opensearch-build/issues/1492 and https://github.com/opensearch-project/opensearch-build/issues/1372 and do not have resources to take this current issue in the near future. Feel free to address it if it is urgent.

**Comment by bbarani** [comment]
@ohltyler @tianleh Do you have any update on this issue?

**Comment by bbarani** [comment]
@ohltyler @tianleh @seraphjiang  Do you have any update on this issue?

**Comment by rishabh6788** [comment]
@ohltyler It seems the cluster verification has been working fine for a while and we have tests to make sure the cluster is up before we start the tests. 
Closing this issue. Feel free to open new issue if you are still seeing flaky errors due to cluster readiness.
